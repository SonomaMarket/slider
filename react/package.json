{
  "name": "vtex.slider",
  "version": "0.0.0",
  "description": "A React Slider component that works well in SPA and SSR",
  "files": [
    "react"
  ],
  "scripts": {
    "test": "vtex-test-tools test",
    "test:watch": "vtex-test-tools test --watch"
  },
  "dependencies": {
    "classnames": "^2.2.6",
    "debounce": "^1.2.0",
    "prop-types": "^15.6.2",
    "ramda": "^0.26.1",
    "react": "^16.6.3",
    "react-event-listener": "^0.6.5"
  },
  "devDependencies": {
    "@vtex/test-tools": "^0.2.0",
    "eslint": "^5.15.3",
    "eslint-config-vtex-react": "^4.0.0",
    "jsdom": "^14.0.0",
    "prettier": "^1.16.4",
    "typescript": "^3.3.4000"
  }
}
