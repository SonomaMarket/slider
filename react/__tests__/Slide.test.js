import React from 'react'
import { render } from '@vtex/test-tools/react'

import Slide from '../components/Slide.js'

describe('<Slide /> component', () => {
  const renderComponent = customProps => {
    const props = {
      ...customProps,
    }

    const comp = (
      <Slide {...props}>
        <div>
          <p>Testing</p>
        </div>
      </Slide>
    )

    return render(comp)
  }

  it('should be rendered', () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toBeTruthy()
  })

  it('should match snapshot', () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })
})
