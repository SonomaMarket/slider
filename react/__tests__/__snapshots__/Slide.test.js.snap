// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slide /> component should match snapshot 1`] = `
<DocumentFragment>
  <li
    class="inline-flex h-100 relative overflow-hidden"
  >
    <div>
      <p>
        Testing
      </p>
    </div>
  </li>
</DocumentFragment>
`;
